src/Storage/ClaimController.ts

```diff
@@ -53,51 +54,142 @@ export class ClaimController {
     const logger = this.logger.child({ method: 'download' })
 
     logger.trace({ ipfsHashes }, 'Downloading Claims')
-
-    await this.collection.insertMany(ipfsHashes.map(ipfsHash => ({ ipfsHash, claimId: null })), { ordered: false })
+    await this.collection.insertMany(
+      ipfsHashes.map((ipfsHash): Entry => ({
+        ipfsHash,
+        claimId: null,
+        lastDownloadAttempt: null,
+        lastDownloadSuccess: null,
+        downloadAttempts: 0,
+      })),
+      { ordered: false }
+    )
   }
 
-  async downloadNextHash() {
-    const logger = this.logger.child({ method: 'downloadNextHash' })
-
-    logger.trace('Downloading Next Hash')
-
-    const ipfsHashEntry = await this.collection.findOne({
-      claimId: null,
-      $or: [{ attempts: { $lt: 5 } }, { attempts: { $exists: false } }],
-    })
-    const ipfsHash = ipfsHashEntry && ipfsHashEntry.ipfsHash
-
-    logger.trace({ ipfsHash }, 'Downloading Next Hash')
-
-    if (!ipfsHash) return
-
-    let claim
-    try {
-      claim = await this.downloadClaim(ipfsHash)
-    } catch (exception) {
-      await this.collection.updateOne({ ipfsHash }, { $inc: { attempts: 1 } })
-
-      logger.debug({ ipfsHash, exception }, 'Failed to Download Claim')
-
-      return
-    }
-
-    logger.info({ ipfsHash, claim }, 'Successfully downloaded claim from IPFS')
-
-    await this.updateClaimIdIPFSHashPairs([
+  async updateEntryPairs({ entry, claim, ...rest }: { claim: Claim; entry: Entry }) {
+    const logger = this.logger.child({ method: 'publishEntryDownload' })
+    logger.trace('started claim update hash pairs')
+    this.updateClaimIdIPFSHashPairs([
```

Comment:
Avoid ignoring returned promises.

If you don't `await` the `this.updateClaimIdIPFSHashPairs`, the promise returned by it will get ignored, which means that any error thrown inside `updateClaimIdIPFSHashPairs` will get ignored too.